tests/e2e/monitoring/test_monitoring_version_mismatch.py

```diff
@@ -0,0 +1,128 @@
+        argnames=['resource', 'alert', 'find_image'],
+        argvalues=[
+            pytest.param(
+                *['rook-ceph-mon-a', 'CephMonVersionMismatch', 'mon:'], marks=pytest.mark.polarion_id("OCS-1571")
+            ),
+            pytest.param(
+                *['rook-ceph-osd-1', 'CephOSDVersionMismatch', 'osd:'], marks=pytest.mark.polarion_id("OCS-1569")
+            ),
+        ]
+    )
+    def test_ceph_version_mismatch(self, resource, alert, find_image):
+        """
+        Test ceph version mismatch
+        """
+        self.resource_name = resource
+        self.find_image = find_image
+        # Get Image name
+        self.original_image = self.find_image_name(resource_name=self.resource_name, find_image=find_image)
+
+        # Check rook ceph version
+        self.original_rook_ceph_version = self.get_rook_ceph_version(name=self.resource_name)
+
+        # Change rook ceph version
+        self.change_image_deployment(image_version='quay.io/ceph-ci/ceph:master', image_type=self.find_image)
+
+        # Check rook ceph version changed
+        rook_ceph_version = self.get_rook_ceph_version(name=self.resource_name)
+        assert rook_ceph_version != self.original_rook_ceph_version,\
+            f"{self.original_rook_ceph_version} is not changed"
+
+        # Look for a Version Mismatch alert in OCP Alerting.
+        prometheus_api = prometheus.PrometheusAPI()
+        alert_records = prometheus_api.wait_for_alert(
+            name=alert, state=None, timeout=15, sleep=3
+        )
+        assert len(alert_records) > 0, f'{alert} alert is not found'
+
+        # Change to original rook ceph version
+        self.change_image_deployment(image_version=self.original_image, image_type=self.find_image)
+
+        # Look for an alert CephOSDVersionMismatch cleared.
+        prometheus_api.check_alert_cleared(
+            label=alert, measure_end_time=60, time_min=120
+        )
+
+    def change_image_deployment(self, image_version, image_type):
+        """
+        Change image deployment
+
+        Args:
+            image_version (str): Change to this version
+            image_type (str): the type of deployment
+
+        """
+        ocp_obj = OCP(namespace='openshift-storage')
+        ocp_obj.exec_oc_cmd(
+            command=f'set image deployment/{self.resource_name} {image_type[:-1]}={image_version}'
+        )
+
+    def find_image_name(self, resource_name, find_image):
+        """
+        Find the Image name of the pod
+
+        Args:
+            resource_name (str): the type of deployment
+            find_image (str): using find Image Name
+
+        Returns:
+            rook_ceph_version (str): the image name
+
+        """
+        ocp_obj = OCP(namespace='openshift-storage', kind='deployment')
+        image = ocp_obj.describe(resource_name=resource_name)
+        image_list = image.split()
+        index = image_list.index(find_image)
+        return image_list[index + 2]
```

Comment:
Why is there `index + 2`? Can you please add comment that describes it?

Reply:
@fbalak 
The output of "oc desscribe" command is string (no yaml)
In this code, I split the output of "oc describe deployment/rook-ceph-mon-a -n openshift-storage" command.
The image name is two places after the "mon:" (in image_list)

Reply:
Thank you. I understand it now. Can you add a short comment about it into code?